### ISSUE TITLE
new temurin 8.0.412+8 is not detected

### ISSUE THREAD

**Issue by hueldoeu** [tushev/aojdk-updatewatcher#75]
**Describe the bug**
i need a good program, that updates my temurin 8 automatically. i found aojdk-updatewatcher and installed it on windows 2012. on 18th april 2024 adoptium published temurin 8.0.412+8 but aojdk-updatewatcher is not finding it.

**I'm sure that it's not a 'known-not-a-bug'**
**Check** https://github.com/tushev/aojdk-updatewatcher/wiki/Known-not-a-bugs
[X] Yes

**To Reproduce**
1. start pc
2. login
3. you can see aojdk-updatewatcher in the task-manager and after a few seconds it disappears.
4. i open aojdk-updatewatcher and see that it can't find updates.
**Expected behavior**
i don't know what would happen if i start the pc and aojdk-updatewatched gets started as scheduled and tells me there is an update.

**Screenshots**
![image](https://github.com/tushev/aojdk-updatewatcher/assets/11273667/4b677979-282c-4eb4-beff-06297d61647e)
![image](https://github.com/tushev/aojdk-updatewatcher/assets/11273667/80f759a6-3704-4a49-8eb2-a6b9c0630a85)

**Desktop (please complete the following information):**
 - App Version? where can i see it?
 - Windows Server 2012
 - Temurin 8.0.402+6

**Additional context**

**Comment by tushev** [comment]
> i don't know what would happen if i start the pc and aojdk-updatewatched gets started as scheduled and tells me there is an update.
> you can see aojdk-updatewatcher in the task-manager and after a few seconds it disappears.

Everything sounds like it's working as expected. When an update will be detected, it will look like this: https://github.com/tushev/aojdk-updatewatcher/raw/master/docs/update_dialog_838.gif?raw=true

**Comment by tushev** [comment]
It seems like an 'upstream' bug. Probably I should switch to a newer API, however, there's no time estimate yet.

**Comment by hueldoeu** [comment]
> > i don't know what would happen if i start the pc and aojdk-updatewatched gets started as scheduled and tells me there is an update.
> > you can see aojdk-updatewatcher in the task-manager and after a few seconds it disappears.
> 
> Everything sounds like it's working as expected. When an update will be detected, it will look like this: https://github.com/tushev/aojdk-updatewatcher/raw/master/docs/update_dialog_838.gif?raw=true
> 
> **EDIT**: by 'working as expected', I mean that [API](https://api.adoptopenjdk.net/v3/assets/latest/8/hotspot?architecture=x64&image_type=jre&os=windows&vendor=adoptopenjdk) (that's outside of my control) returns `8.0.402+6`.

who controls the API? as i can see on the link it's still the older version. how can i help to update the API? if it gets updates up in between 14 days, then it's ok.

**Comment by tushev** [comment]
Eclipse Foundation > Adoptium Project. I opened an issue there, it's [linked above](https://github.com/adoptium/api.adoptium.net/issues/1014).

> how can i help to update the API? 

Thank you, but there's nothing you can help me with.

**Comment by hueldoeu** [comment]
![image](https://github.com/tushev/aojdk-updatewatcher/assets/11273667/1ae4f531-3c0d-4c25-93a1-4bf50b054279)
also not updated on winget
Eclipse Temurin JRE with Hotspot 8  EclipseAdoptium.Temurin.8.JRE  8.0.402.6  Tag: temurin    winget
if winget doesn't update properly then it has no way out.
win server 2025 will include winget and i really enjoy it on windows 10/11

**Comment by hueldoeu** [comment]
will Update Watcher for AdoptOpenJDK be updated on its own? can it update itself?

**Comment by hueldoeu** [comment]
cool :)
![image](https://github.com/tushev/aojdk-updatewatcher/assets/11273667/86ba3def-9826-4eff-ba44-52e594c08058)
it feels good seeing this. i am having big problems in LMU munich vet with students all over the campus including most docents.
this windows of Update Watcher for AdoptOpenJDK gave me a smile. i really enjoyed it getting splashed by this window after login.

**Comment by hueldoeu** [comment]
i'm thinking of starting the shedule even before someone has logged in because we use the server only once per week
![image](https://github.com/tushev/aojdk-updatewatcher/assets/11273667/bebf3df3-3ff3-4ad6-a7d9-83f18b4bbb33)
![image](https://github.com/tushev/aojdk-updatewatcher/assets/11273667/50c6f301-21a3-457c-83f7-69ecc3130b19)
![image](https://github.com/tushev/aojdk-updatewatcher/assets/11273667/daa9af79-1471-4891-b4f7-c24103a13797)
maybe you can also add some extra arguments like "auto-install if update is available" but according to my experiences with unigetUI or related software, it's better to keep an eye on the update routine
![image](https://github.com/tushev/aojdk-updatewatcher/assets/11273667/9a08bc1d-81a8-4d9f-a10a-f6845bdf4f47)

**Comment by tushev** [comment]
It seems that Adoptium fixed something on their side, as I did nothing :)

I'm planning to switch to a newer API, however, this is not a simple 'change URL' task because Temurin builds are available on the new API but OpenJ9/Semeru remain on the old API. A lot of code has to be changed, and I'm a bit busy, so there's no ETA for that.

> i'm thinking of starting the shedule even before someone has logged in because we login on the server only once per week, server starts am 6 am and shuts down at 11 pm mo-fr

Yes, an advanced user can change the task as required. Or, better, disable it and create a separate task because default one (`Adopt_UpdateWatcher_TskV2`) may be changed automatically and thus revert to default settings.

> maybe you can also add some extra arguments like "auto-install if update is available" but according to my experiences with unigetUI or related software, it's better to keep an eye on the update routine

I'm planning to add MSI flags for quiet/silent update at some point (#10) so the end user won't have to go through clicking the MSIs.

As for auto-install - I'm against this because version detection frequently [breaks](https://github.com/tushev/aojdk-updatewatcher/issues/72#issuecomment-1818515855) due to unexpected changes on Adoptium side. Having an active auto-install feature will result in updating the same instance over and over.

**Comment by hueldoeu** [comment]
after login this morning, another update appeared: 8.0.422+5.

update procedure was delightful! ♥

however we MUST take distance of windows server 2012 so one day aojdk-updatewatcher can be obsolete maybe (cmd => winget upgrade)
